R/include-hba1c.R

```diff
@@ -0,0 +1,32 @@
+#' Include only those with HbA1c in the required range.
+#'
+#' In the `lab_forsker` register, NPU27300 is HbA1c in the modern units (IFCC)
+#' while NPU03835 is HbA1c in old units (DCCT).
+#'
+#' @param data The `lab_forsker` register.
+#'
+#' @return An object of the same input type, default as a [tibble::tibble()],
+#'   with two columns: `pnr` and `included_hba1c`.
+#' @keywords internal
+#'
+#' @examples
+#' register_data$lab_forsker |> include_hba1c()
+include_hba1c <- function(data) {
+  verify_required_variables(data, "lab_forsker")
+  hba1c_criteria <- get_algorithm_logic("hba1c")
+  data |>
+    column_names_to_lower() |>
+    dplyr::filter({{ hba1c_criteria }}) |>
+    # Keep only the columns we need.
+    dplyr::transmute(
+      pnr = .data$patient_cpr,
+      date == .data$samplingdate,
+      included_hba1c = TRUE
```

Comment:
So this step outputs a df with three columns (`pnr`, `date`, and `included_hba1c`) where `included_hba1c` is TRUE for all rows, since this only includes individuals with samples over the threshold?

Reply:
Almost. This step creates a column with only the existing pnr and date, then adds a new column called `included_hba1c` where the only value of the column is TRUE

Reply:
Maybe consider using mutate(.keep = "none") since transmute is superseded according to https://dplyr.tidyverse.org/reference/transmute.html